### PR TITLE
improve user api performance

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?183105

adds DB-level pagination and switches to stale views.

the change is in https://github.com/dimagi/commcare-hq/commit/12403571eb65c040411bd71056cbb629f8dec094 everything else was prework

@dannyroberts

### CODE REVIEW

File: corehq/apps/api/resources/v0_4.py [hunk 1]
```diff
@@ -217,9 +217,9 @@ def obj_get_list(self, bundle, domain, **kwargs):
         if 'size' in query:
             del query['size']
 
-        return ESQuerySet(payload = query,
-                          model = CommCareCase,
-                          es_client = self.case_es(domain)).order_by('server_modified_on') # Not that CaseES is used only as an ES client, for `run_query` against the proper index
+        return ESQuerySet(payload=query,
+                          model=CommCareCase,
+                          es_client=self.case_es(domain)).order_by('server_modified_on') # Not that CaseES is used only as an ES client, for `run_query` against the proper index
```
Comment (dimagimon) [inline corehq/apps/api/resources/v0_4.py]: E261 at least two spaces before inline comment

Comment (dimagimon) [inline corehq/apps/api/resources/v0_4.py]: E501 line too long (177 > 115 characters)

**Comment by czue** [comment]
:white_check_mark:

**Comment by snopoke** [comment]
nice

**Comment by benrudolph** [comment]
crushing it

**Pull request merged**